### ISSUE TITLE
Check the impact on the limits of trigger matching either of the two highest ionizing HSCPs to trigger object and corresponding tracker or global muon.

### ISSUE THREAD

**Issue by gracecummings** [CMS-HSCP/SUSYBSMAnalysis-HSCP#100]

**Comment by tvami** [comment]
@gracecummings so the idea is
 * we selected the highest Ih track --> check if it's the one triggered --> if not, discard that event?

**Comment by gracecummings** [comment]
No, I would check if the trigger object matches any (or at least either of
the two highest pT) HCSP candidates in the event. If there are two (like
the staus), either could fire the trigger.

If then it does not match, discard the event. The idea (IMO) is to probe
the effect of that 5% loss of efficiency shown in figure 38 on the final
sensitivity.

**Comment by tvami** [comment]
Ok, but generally this will mean we select the **less ionizing** one, since that will be faster and so more likely to reach the trigger, right?

**Comment by gracecummings** [comment]
Sure, but I do not see why that would be a problem. As long as we understand what is firing the trigger, and the origin of the object, we can correct for mismodeling (if it is needed).

**Comment by lawrenceleejr** [comment]
I think these are wide distributions, so I there might be a small bias in that direction. But the point is it doesn't matter so much which triggered it -- we then have the freedom to use whatever we want in the actual analysis once we're happy that the trigger did its job and that we are reasonably modeling it.

But @gracecummings -- I'm not sure this goes in the right direction for probing that efficiency loss. The thought behind what we're currently doing was:

We just want to remove events where the trigger has fired on something that we don't want it to have considered a muon. So we just need to clean up the trigger by requiring something well-measured offline. So we're currently just requiring that a "good" offline muon is consistent with the HLT muon. As long as that's the case, we can trust the trigger and don't necessarily care about the relationship with the HSCP candidates. This helps us be more inclusive of other signals. E.g. it's easy to imagine associated production of a muon and a heavy lepton-like thing. Or V+HSCPs, or whatever. Treating the trigger matching in this way ensures we're being as inclusive as we can be (given the trigger choice) but still have a solid understanding of what fired the trigger and that it was a reasonably well measured object.

**Comment by gracecummings** [comment]
@lawrenceleejr I agree with you that as long as we have a well modeled and understood trigger object, that we should be OK. One could just apply basic muon trigger scale factors and be done with it.

The point of checking the matching to the HSCP (either in addition to matching a muon, or just matching straight) is we want to understand how the signal triggers. If in the events where the matched muon does not come is not an HSCP, but comes from something else (like being pulled out of an ISR jet -- no idea if this is possible), we cannot be sure that the trigger modeling in the signal samples is something that is well understood, and by proxy, have a systematic assigned. 

Especially with the way the uncertainty is done currently with respect to the beta of the HSCP. If the trigger does not come from the HSCP, we are actually not quantifying how likely that event was triggered properly. This of course can be accounted for with the muon trigger scale factors for the muon, but that does not say anything about how the muon modeling is in the event. I will admit, it also is not clear if this is needed, but OR will bring it up.

Maybe my language was misleading, but I meant this to explore the effect on the analysis of taking the hit in efficiency through the matching. This check was requested by both convenors, so it needs to be done anyway. If the impact is negligible, it would greatly simplify the uncertainties coming from the trigger.

**Comment by tvami** [comment]
> like being pulled out of an ISR jet 

I found some events where this happens, it's very very rare, but it happens

**Comment by tvami** [comment]
Relevant plots were shown today
https://indico.cern.ch/event/1257416/#5-news-from-jhu
